vagrant/src/main/java/org/jclouds/vagrant/internal/VagrantCliProvider.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.vagrant.internal;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Collection;
+
+import javax.inject.Inject;
+
+import org.jclouds.domain.LoginCredentials;
+import org.jclouds.vagrant.api.VagrantApiFacade;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.io.Files;
+
+import vagrant.Vagrant;
+import vagrant.api.CommandIOListener;
+import vagrant.api.VagrantApi;
+import vagrant.api.domain.Box;
+import vagrant.api.domain.SshConfig;
+
+public class VagrantCliProvider implements Function<File, VagrantApiFacade<Box>> {
+
+   static class VagrantCliFacade implements VagrantApiFacade<Box> {
+
+      private final VagrantApi vagrant;
+      private final VagrantOutputRecorder outputRecorder;
```

Comment:
Agree a factory would be better suited. That's a nice trick, I've implemented it.